### DISCUSSION TITLE
Pretty print in production

### DISCUSSION THREAD

**Discussion by artplan1** [palkan/action_policy#145] [Q&A]
Hi @palkan,
I was checking required memory in our project using derailed (`RAILS_ENV=production bundle exec derailed bundle:mem`) and saw this:

```ruby
action_policy: 4.8359 MiB (Also required by: action_policy/graphql)
    action_policy/base: 4.7617 MiB
      action_policy/policy/core: 4.7539 MiB
        action_policy/utils/pretty_print: 4.6602 MiB
          parser/current: 4.582 MiB
            parser/ruby27: 4.5469 MiB
              parser: 2.9453 MiB
                parser/lexer: 2.7461 MiB
```

It seems like pretty print is always required (which is strange for production envs). Or it's how derailed outputs info?

**Comment by palkan** [comment]
> It seems like pretty print is always required

it is. But it's no-op unless Parser, Unparser and MethodSource are present in the bundle.
So, the real question is: what `parser` is doing in production env?

**Reply by artplan1** [reply]
okay, I checked Gemfile.lock. `parser` is required by rubocop, but it's in development group and has `require: false`. If I comment `rubocop` in gemfile - I don't see `action_policy` in derailed output anymore.
I assume it's not an issue then.
Thank you!

**Reply by palkan** [reply]
> it's in development group and has require: false

Do you run `bundle install --without development test` in production or just `bundle install`?
`require: false` only tells Rails not load the gem automatically but it's still available to the application (and installed).

**Reply by artplan1** [reply]
sure, it's installed without dev/test deps in production

I checked it locally, that's why I saw such output. Just retried locally to reinstall gems with `--without development test` and all looks good `action_policy: 0.4922 MiB`

in derailed docs it says `By default this task will only return results from the :default and "production" groups`, that's why I was confused :)
